build_config/lib/src/key_normalization.dart

```diff
@@ -2,13 +2,25 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-String normalizeBuilderKey(String builderKey, String packageName) {
+String normalizeBuilderKeyDefinition(String builderKey, String packageName) {
```

Comment:
I am a bit confused by all of these and exactly what they correspond to - could you add comments with a quick example of how they correspond to different usages in the yaml?

Reply:
Added examples.